Update partnerships.mdx (#2319)
hashicorp/web-unified-docs: content/terraform-docs-common/docs/docs/partnerships.mdx

**Feature:**
- [Ephemeral Resources](https://developer.hashicorp.com/terraform/plugin/framework/ephemeral-resources) if applicable as a security best practice
- Terraform Search, which requires the implementation of both [resource identities](https://developer.hashicorp.com/terraform/plugin/framework/resources/identity) and [list resources](https://developer.hashicorp.com/terraform/plugin/framework/list-resources)
- [Terraform Actions](https://developer.hashicorp.com/terraform/plugin/framework/actions)

Technology partners that qualify for the Partner Premier tag enjoy additional partner benefits, such as co-marketing support and opportunities, technical support and resources, roadmap sharing, access to provider development workshops, and exclusive design input into the future of Terraform as a product, in addition to other perks.

#### Requirements for the Partner Tag

> Moving forward, in order to qualify for the Partner tag on the Terraform Registry, the following must be met and verified by a HashiCorp-IBM Product Partnerships Manager:

Reviewer: ```suggestion
To qualify for the Partner tag on the Terraform Registry, a HashiCorp-IBM Product Partnerships Manager must verify that your provider meets the following requirements:
```

- Adhere to [Terraform’s naming conventions](https://developer.hashicorp.com/terraform/plugin/best-practices/naming) for resources, variables, and outputs.
- Module format should be checked using terraform fmt to ensure code follows standard format.
- Include the following files in your module:
  - **main.tf**: The core configuration file where resources are defined. 
  - **variables.tf**: Stores input variable, allowing operators to parameterize Terraform configurations.
  - **outputs.tf**: Defines output values, making data from infrastructure available to other configurations or users.
  - **README.md**: Provides a project overview, instructions, and documentation. 

>   - **Changelog**: Documents changes and updates to the module over time.

Reviewer: ```suggestion
  - `main.tf`: The core configuration file where resources are defined. 
  - `variables.tf`: Stores input variable, allowing operators to parameterize Terraform configurations.
  - `outputs.tf`: Defines output values, making data from infrastructure available to other configurations or users.
  - `README.md`: Provides a project overview, instructions, and documentation. 
  - Changelog.txt: Documents changes and updates to the module over time.
```
.txt for the Changelog?
